components/apimgt/org.wso2.carbon.apimgt.rest.api.core/src/main/java/org/wso2/carbon/apimgt/rest/api/core/utils/PolicyExportManager.java

```diff
@@ -0,0 +1,242 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.core.utils;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.core.api.APIMgtAdminService;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.exception.APIMgtDAOException;
+import org.wso2.carbon.apimgt.core.models.policy.APIPolicy;
+import org.wso2.carbon.apimgt.core.models.policy.ApplicationPolicy;
+import org.wso2.carbon.apimgt.core.models.policy.CustomPolicy;
+import org.wso2.carbon.apimgt.core.models.policy.SubscriptionPolicy;
+import org.wso2.carbon.apimgt.core.template.APITemplateException;
+import org.wso2.carbon.apimgt.core.template.APIThrottlePolicyTemplateBuilder;
+import org.wso2.carbon.apimgt.core.template.ApplicationThrottlePolicyTemplateBuilder;
+import org.wso2.carbon.apimgt.core.template.CustomThrottlePolicyTemplateBuilder;
+import org.wso2.carbon.apimgt.core.template.SubscriptionThrottlePolicyTemplateBuilder;
+import org.wso2.carbon.apimgt.core.util.APIFileUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Policy export manager handles creating execution plans and creating and archive.
+ */
+public class PolicyExportManager {
+
+    private static final Logger log = LoggerFactory.getLogger(PolicyExportManager.class);
+    private static final String APPLICATION = "application_";
+    private static final String SUBSCRIPTION = "subscription_";
+    private static final String RESOURCE = "resource_";
+    private static final String DEFAULT = "_default";
+    private static final String CUSTOM = "custom_";
+    private static final String ZIP = ".zip";
+    private static final String EXPORT_POLICIES = "ExportPolicies";
+    private APIMgtAdminService apiMgtAdminService;
+
+    public PolicyExportManager(APIMgtAdminService adminService) {
+        apiMgtAdminService = adminService;
+    }
+
+    public String createArchiveFromExecutionPlans(String exportedPoliciesDirName, String archiveDir, String archiveName)
+            throws APIManagementException {
+
+        try {
+            //retrieve all policies under each policy level
+            List<APIPolicy> apiPolicies = apiMgtAdminService.getApiPolicies();
+            List<ApplicationPolicy> applicationPolicies = apiMgtAdminService.getApplicationPolicies();
+            List<SubscriptionPolicy> subscriptionPolicies = apiMgtAdminService.getSubscriptionPolicies();
+            List<CustomPolicy> customPolicies = apiMgtAdminService.getCustomRules();
+            //write all execution Plans/Siddhi Apps to exportPoliciesDirName directory
+            String rootFolder = exportedPoliciesDirName;
```

Comment:
we modify exportedPoliciesDirName later such that we add another folder inside it. Therefore to keep the root folder I did this. But I made a change as you requested.